### PR TITLE
executor: improve trace format='row'

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

In https://github.com/pingcap/tidb/pull/8357, the trace result is in json format, used by web UI.

This commit add a row format for it, which look like this:

```
mysql> trace format='row' select * from t1 order by c limit 3;
+---------------------------+---------------------------+------------+
| operation                 | startTS                   | duration   |
+---------------------------+---------------------------+------------+
| session.getTxnFuture      | Jan 11 21:33:09.933212243 | 4.181µs    |
|   ├─session.Execute       | Jan 11 21:33:09.933207819 | 763.708µs  |
|   ├─session.ParseSQL      | Jan 11 21:33:09.933293211 | 57.002µs   |
|   ├─executor.Compile      | Jan 11 21:33:09.933388076 | 233.427µs  |
|   ├─session.runStmt       | Jan 11 21:33:09.933658842 | 198.985µs  |
|   ├─session.CommitTxn     | Jan 11 21:33:09.933808415 | 11.947µs   |
|   ├─recordSet.Next        | Jan 11 21:33:09.934038447 | 691.335µs  |
|   ├─topN.Next             | Jan 11 21:33:09.934042362 | 663.15µs   |
|   ├─tableReader.Next      | Jan 11 21:33:09.934050669 | 548.271µs  |
|   ├─tableReader.Next      | Jan 11 21:33:09.934671187 | 5µs        |
|   ├─recordSet.Next        | Jan 11 21:33:09.934756161 | 22.416µs   |
|   └─topN.Next             | Jan 11 21:33:09.934760051 | 1.553µs    |
+---------------------------+---------------------------+------------+
12 rows in set (0.00 sec)
```

### What is changed and how it works?

Write the trace result to chunk in row format.

There are some small pitfalls I'll investigate later:
1. The opentracing span parent-children relationship seems incorrect 
2. Add more columns to provide useful info

We support `trace format='json' ... ` and `trace format='row' ...`
Maybe I should also make format=row the default one.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @lysu

File: executor/trace.go [hunk 1]
```diff
@@ -189,11 +167,19 @@ func dfsTree(span basictracer.RawSpan, tree map[uint64][]basictracer.RawSpan, pr
...
+	var start time.Time
+	var duration time.Duration
+	if e, err := t.TimespanEvent(); err == nil {
+		start = e.Start()
+		end := e.End()
+		duration = end.Sub(start)
+	}
+
+	chk.AppendString(0, prefix+suffix+t.Span.Name())
+	chk.AppendString(1, start.Format("15:04:05.000000"))
```
Comment (jackysp) [inline executor/trace.go]: It is OK for me, but "15:04:05.000000" is still a little strange.

Comment (tiancaiamao) [reply executor/trace.go]: This control the time format output.
Go use pattern match to control the output format, https://godoc.org/time#Parse
And the layout string always use this date: 2006 01 02 15 04 05, it's a special number

if you want the result looks like "hh:mm:ss.xxxxxx" , the layout should be "15:04:05.000000"

**Comment by lysu** [comment]
LGTM to current code, but maybe more refine the span name in the next @tiancaiamao 

```
mysql> trace format="row" select * from t join t2 on t2.id = t.id where t.id = 2;
+---------------------------+-----------------+------------+
| operation                 | startTS         | duration   |
+---------------------------+-----------------+------------+
| session.getTxnFuture      | 12:02:26.316169 | 3.014µs    |
|   ├─session.Execute       | 12:02:26.316166 | 867.827µs  |
|   ├─session.ParseSQL      | 12:02:26.316213 | 54.937µs   |
|   ├─executor.Compile      | 12:02:26.316297 | 379.766µs  |
|   ├─session.runStmt       | 12:02:26.316714 | 272.396µs  |
|   ├─session.CommitTxn     | 12:02:26.316955 | 9.873µs    |
|   ├─recordSet.Next        | 12:02:26.317053 | 449.804µs  |
|   ├─tableReader.Next      | 12:02:26.317115 | 324.444µs  |
|   └─tableReader.Next      | 12:02:26.317125 | 288.022µs  |
+---------------------------+-----------------+------------+
9 rows in set (0.01 sec)
```
from this two table with same name we don't know which table have question, also miss join execution info, and maybe trace insert is useful too.

**Comment by lysu** [comment]
/run-common-test tidb-test=pr/715
/run-integration-common-test=pr/715

**Pull request merged**